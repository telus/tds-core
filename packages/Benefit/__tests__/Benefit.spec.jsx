import React from 'react'
import { shallow, render } from 'enzyme'

import BenefitItem from '../BenefitWithHeading/BenefitWithHeadingItem'

describe.skip('Benefit', () => {
  const doShallow = (props = {}) => shallow(<BenefitItem {...props} />)

  it('renders', () => {
    const benefit = render(<BenefitItem />)

    expect(benefit).toMatchSnapshot()
  })

  it('does other things', () => {
    const benefit = doShallow()

    expect(benefit).toExist()
  })

  it('renders an HTML ul tag', () => {
    const benefit = doShallow()

    expect(benefit.dive()).toHaveDisplayName('ul')
  })

  it.skip('list items render an HTML li tag', () => {
    const benefitItem = doShallow(<BenefitItem.Item>Some content</BenefitItem.Item>)

    expect(benefitItem).toHaveDisplayName('li')
  })

  it('passes additional attributes to the element', () => {
    const benefit = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(benefit).toHaveProp('id', 'the-id')
    expect(benefit).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const benefit = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(benefit).not.toHaveProp('className', 'my-custom-class')
    expect(benefit).not.toHaveProp('style')
  })
})
