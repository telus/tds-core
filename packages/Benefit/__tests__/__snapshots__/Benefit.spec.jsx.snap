// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefit renders 1`] = `
<ul
  class="TDS_Box-modules__betweenBottomMargin-3___1jBrz TDS_Box-modules__stack___33m4D"
/>
`;
